File: 2559-count-vowel-strings-in-ranges/2559-count-vowel-strings-in-ranges.py
class Solution:
    def vowelStrings(self, words: List[str], queries: List[List[int]]) -> List[int]:
        n = len(words)
        v = [0] * (n + 1)
        vowel = 'aeiou'
        ans = []
        for i in range(n):
            v[i + 1] = v[i] + (1 if words[i][0] in vowel and words[i][-1] in vowel else 0)
        for l, r in queries:
            ans.append(v[r + 1] - v[l])
        return ans

File: 2559-count-vowel-strings-in-ranges/2559-count-vowel-strings-in-ranges.cpp
class Solution {
public:
    bool is(char x){
    if (x == 'a' || x == 'e' || x == 'i' ||x == 'o' || x == 'u')
        return true;
    else return false;
}
    vector<int> vowelStrings(vector<string>& arr, vector<vector<int>>& q) {
        vector<int> ans;
        vector<int>prefix(arr.size(),0);
        if(is(arr[0][0]) and is(arr[0][arr[0].size()-1]))prefix[0]=1;
        for(int i=1;i<arr.size();i++){
            if(is(arr[i][0]) and is(arr[i][arr[i].size()-1])){prefix[i]+=prefix[i-1]+1;}
            else prefix[i]=prefix[i-1];
        }
        for(int i=0;i<q.size();i++){
            int l=q[i][0],r=q[i][1];
            int cnt=0;
            if(l==0) ans.push_back(prefix[r]);
            else ans.push_back(prefix[r]-prefix[l-1]);
        }
        return ans;
    }
};